### PR TITLE
Insert single data item instead of array for better error handling

### PR DESCRIPTION
**Pull request by arajkumar**
* Access gremlin server directly
* Add more tests
* Fixed creation of dummy node when adding feedback for non existent SecurityEvent

### CODE REVIEW

File: src/feedback.py [hunk 1]
```diff
@@ -13,16 +13,12 @@ def _ingest_feedback(payload):
     security_event = SecurityEvent(url=payload['url'])
     feedback_ = Feedback(author=payload['author'],
                          feedback_type=FeedBackType[payload['feedback_type']],
+                         url=payload['url'],
                          comments=payload['comments'])
-    g.has_node(security_event).add_unique_node(feedback_)
+    g.has_node(security_event).and_(Traversel.anonymous().add_unique_node(feedback_))
```
Comment (rajusem) [inline src/feedback.py]: you can think of  handling  exception in all graph db call Traversel class.

File: requirements.txt [hunk 2]
```diff
@@ -25,20 +22,16 @@ isodate==0.6.0            # via gremlinpython
 itsdangerous==1.1.0       # via flask
 jinja2==2.10.3            # via flask
 jsonschema==3.2.0         # via flask-restplus
-locust==0.0
-locustio==0.13.2          # via locust
 markupsafe==1.1.1         # via jinja2
 more-itertools==8.0.2     # via zipp
-msgpack-python==0.5.6     # via locustio
-https://github.com/fabric8-analytics/osa-data-model/archive/master.zip
```
Comment (rajusem) [inline requirements.txt]: So now we are not using osa-data-model anywhere ?

File: src/gremlin.py [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+import json
+import logging
+import requests
+import daiquiri
 from gremlin_python.structure.io import graphsonV3d0
+from src.config import GREMLIN_DEFAULT_HOST, GREMLIN_DEFAULT_PORT, GREMLIN_SCHEME
 
-from gremlin_connect.gremlin_adapter import GremlinAdapter
-from src.config import GREMLIN_DEFAULT_HOST, GREMLIN_DEFAULT_PORT
+daiquiri.setup(level=logging.DEBUG)
```
Comment (rajusem) [inline src/gremlin.py]: Can you read  this logging mode from config file so if we wanted to change to INFO then can change config without changing code.

File: src/feedback.py (hunk 1 above)
Comment (rajusem) [inline src/feedback.py]: Also think of adding retry mechanism in case of any failure.

Comment (arajkumar) [reply src/feedback.py]: I'm not anticipating any handleable exception over here.

File: requirements.txt (hunk 2 above)
Comment (arajkumar) [reply requirements.txt]: Yes. We were only using gremlin adapter earlier., that has been removed to have a better control over error handling.

File: src/gremlin.py (hunk 3 above)
Comment (arajkumar) [reply src/gremlin.py]: Sure. I can add a env variable to control the level.

File: src/feedback.py (hunk 1 above)
Comment (arajkumar) [reply src/feedback.py]: I will create a separate PR to add retry mechanism once we decide the approach.

**Comment by rootAvish** [comment]
I think the traversal here will significantly increase the time it takes to add a node, am I wrong about that?

**Comment by arajkumar** [comment]
> I think the traversal here will significantly increase the time it takes to add a node, am I wrong about that?

No, It won't. As discussed, it generates Gremlin query as string and executes it through gremlin REST api.

**Pull request merged**